src/simplify_algebra.cpp

```diff
@@ -421,11 +435,13 @@ struct find_dot_mul
 struct find_mul_add
 {
     auto matcher() const
-    {
-        return match::name("mul")(match::either_arg(0, 1)(
+    {   
+        auto slice_1 = match::skip(match::name("slice"))(match::none_of(fusable_split("add")));
```

Comment:
This should be `match::none_of(match::name("slice")(match::arg(0)(fusable_split("add"))))`.